frontend/ui/widget/keyvaluepage.lua

```diff
@@ -543,6 +547,100 @@ function KeyValuePage:_populateItems()
     self.return_button:resetLayout()
     self.main_content:clear()
     local idx_offset = (self.show_page - 1) * self.items_per_page
+
+    -- for flexible middle ratio calculation
+    -- in sync with KeyValueItem actual computation
+    local frame_padding = KeyValueItem.frame_padding
+    local frame_internal_width = self.item_width - frame_padding * 2
+    local middle_padding = KeyValueItem.middle_padding
+    local available_width = frame_internal_width - middle_padding
+    -- Default widths (and position of value widget) if each text fits in 1/2 screen width
+    local key_w = math.floor(frame_internal_width / 2 - middle_padding)
+    local value_w = math.floor(frame_internal_width / 2)
```

Comment:
As an aside, I think @zwim would want to write this as:
```suggestion
    local key_w = math.floor(frame_internal_width * (1/2) - middle_padding)
    local value_w = math.floor(frame_internal_width * (1/2))
```